app/models/concerns/user_totp_methods.rb

```diff
@@ -37,6 +38,7 @@ def enable_totp!(seed, level)
     if mfa_device_count_one?
       self.mfa_level = level
       self.mfa_recovery_codes = Array.new(10).map { SecureRandom.hex(6) }
+      self.hashed_mfa_recovery_codes = mfa_recovery_codes.map { |code| BCrypt::Password.create(code) }
```

Comment:
Just a note - There is also the [cost factor](https://github.com/bcrypt-ruby/bcrypt-ruby#cost-factors) stuff that can be messed with. Lower cost factor is faster to compute, but less secure. Haven't messed with this locally yet, but if speed is problematic, we can toy with it a little bit.